test/export/test_export.py

```diff
@@ -3744,17 +3744,91 @@ def forward(self, x):
         for op in ops:
             self.assertIn(op, (torch.ops.aten._to_copy.default,))
 
-    def test_device_to_mutation(self):
+    def test_device_to_mutation_return(self):
         class Module(torch.nn.Module):
             def forward(self, x):
                 y = x.to("cpu")
                 y.add_(1)
                 return y, x
 
+        torch.export.export_for_training(Module(), (torch.tensor(1, device="cpu"),))
+
         with self.assertRaisesRegex(
-            RuntimeError, "cannot mutate tensors with frozen storage"
+            RuntimeError, "Cannot return on mutated frozen tensor"
         ):
-            export(Module(), (torch.tensor(1, device="cpu"),))
+            torch.export.export_for_inference(
+                Module(), (torch.tensor(1, device="cpu"),)
+            )
+        # with self.assertRaisesRegex(
+        #     RuntimeError, "Cannot return on mutated frozen tensor"
+        # ):
+        #     export(
+        #         Module(), (torch.tensor(1, device="cpu"),)
+        #     )
+
+    def test_device_to_mutation_return_float(self):
+        class Module(torch.nn.Module):
+            def forward(self, x):
+                y = x.float()
+                y.add_(1)
+                return y, x
+
+        torch.export.export_for_training(Module(), (torch.tensor(1, device="cpu"),))
```

Comment:
Today there is some problem with how this to logic is preserved in training IR. Could you also test this for torch.export.export?

Reply:
> Today there is some problem with how this to logic is preserved in training IR. Could you also test this for torch.export.export?

Not sure if this is a comment on the outdated version. I just exported to gh again. All tests should have both export_for_training and export.export now.

Reply:
Ok sounds good :)